apps/condo/domains/organization/utils/accessSchema.js

```diff
@@ -1,239 +1,308 @@
-async function checkPermissionInUserOrganizationOrRelatedOrganization (userId, organizationId, permission) {
-    return checkPermissionsInUserOrganizationOrRelatedOrganization(userId, organizationId, [permission])
+ * @returns {Promise<void>}
+ */
+async function resetUserEmployeesCache (userId) {
+    const cacheKey = _getUserOrganizationsCacheKey({ id: userId })
+    await _redisClient.del(cacheKey)
 }
 
 /**
- * @param {string} userId
- * @param {string} organizationId
- * @param {string[]} permissions
- * @return {Promise<boolean>}
+ * Information about single organization, in which user is employed, and its child organizations
+ * @typedef {Object} UserOgranizationInfo
+ * @property {Record<string, boolean>} permissions - user permissions in organization
+ * @property {Array<string>} childOrganizations - list of child organizations to which user share permissions via OrganizationLink
  */
-async function checkPermissionsInUserOrganizationOrRelatedOrganization (userId, organizationId, permissions) {
-    if (!userId || !organizationId) return false
 
-    const hasPermissionsInUserOrganization = await checkOrganizationPermissions(userId, organizationId, permissions)
-    if (hasPermissionsInUserOrganization) return true
-    return await checkRelatedOrganizationPermissions(userId, organizationId, permissions)
-}
+/**
+ * Information about all organizations user has access to
+ * @typedef {Object} UserOgranizationsCache
+ * @property {number} dv - cache entry data version
+ * @property {Record<string, UserOgranizationInfo>} organizations - record where keys are organization ids and values are info it
+ */
 
 /**
- * Check that the user has access in each organization
- *
- * @param userId {string} User.id field
- * @param organizationIds {Array<string>} array of objects related organizations
- * @param permission {string} OrganizationEmployeeRole permission key to check for
- * @return {Promise<boolean>}
+ * Obtains user organizations info via caching or sub-querying
+ * @param {{ id: string }} user - user object
+ * @returns {Promise<UserOgranizationsCache>}
+ * @private
  */
-async function checkOrganizationsPermission (userId, organizationIds, permission) {
-    if (!userId || !isArray(organizationIds) || isEmpty(organizationIds)) return false
+async function _getUserOrganizations (user) {
+    const cacheKey = _getUserOrganizationsCacheKey(user)
+
+    if (!DISABLE_USER_ORGANIZATION_CACHING) {
+        const cachedDataString = await _redisClient.get(cacheKey)
 
-    const uniqOrganizationIds = uniq(organizationIds)
+        if (cachedDataString !== null) {
+            return JSON.parse(cachedDataString)
+        }
+    }
+
+    const newCacheEntry = { dv: 1, organizations: {} }
+
+    const userEmployees = await find('OrganizationEmployee', {
+        deletedAt: null,
+        organization: { deletedAt: null },
+        user: { id: user.id },
+        isAccepted: true,
+        isBlocked: false,
+        isRejected: false,
```

Comment:
Do we really need to check that it’s Rejected is false? As I remember we have such legacy  cases. But I’m not sure

Reply:
Idk, all tests are passed multiple times...So I guess there's no more cases like that (otherwise someone would fix this behaviuor by test)